packages/core/src/editor/query.tsx

```diff
@@ -27,60 +29,47 @@ import {
   ERROR_INVALID_NODE_ID,
 } from "@craftjs/utils";
 import findPosition from "../events/findPosition";
+import { mergeTrees } from "../utils/mergeTrees";
 import { getDeepNodes } from "../utils/getDeepNodes";
 import { transformJSXToNode } from "../utils/transformJSX";
 
-export function QueryMethods(Editor: EditorState) {
-  const options = Editor && Editor.options;
+export function QueryMethods(state: EditorState) {
+  const options = state && state.options;
 
   const _: () => QueryCallbacksFor<typeof QueryMethods> = () =>
-    QueryMethods(Editor);
+    QueryMethods(state);
 
   const getNodeFromIdOrNode = (node: NodeId | Node) =>
-    typeof node === "string" ? Editor.nodes[node] : node;
+    typeof node === "string" ? state.nodes[node] : node;
 
   return {
-    /**
-     * Get the current Editor options
-     */
-    getOptions(): Options {
-      return options;
-    },
     /**
      * Get a Node representing the specified React Element
-     * @param child
+     * @param reactElement
      * @param extras
      */
-    createNode(child: React.ReactElement | string, extras?: any) {
-      const node = transformJSXToNode(child, extras);
+    createNode(reactElement: React.ReactElement | string, extras?: any) {
+      const node = transformJSXToNode(reactElement, extras);
+
       const name = resolveComponent(options.resolver, node.data.type);
       invariant(name !== null, ERRROR_NOT_IN_RESOLVER);
-      node.data.displayName = node.data.displayName
-        ? node.data.displayName
-        : name;
-
+      node.data.displayName = node.data.displayName || name;
```

Comment:
This will use the name from the resolver always, but we want to give priority to the name specified in the `.craft` property (which is stored in `.data.displayName`).

Reply:
hmm I'm a bit confused.
I didn't change the logic here, I just removed a useless ternary for a more readable or statement. Isn't this statement doing what you want it to do?

Reply:
Oops sorry my bad, I got mixed up 😆